### ISSUE TITLE
Can we please get soft assertions added to the python tools

### ISSUE THREAD

**Issue by heathiviesyncro** [microsoft/playwright-python#1272]
It would be nice to get the soft assertions that are available in node to be available in python.

https://playwright.dev/docs/test-assertions#soft-assertions

Thank you

**Comment by aspenboy** [comment]
I am using `pytest-check` for soft assertions. Adds another dependency to the project, but it works great.

**Comment by jfp1992** [comment]
> pytest-check

I tried this one but couldn't get it working with expect(). Do you have an example of this?

Also +1 for soft assertions to be implemented in python as well as typescript.

**Comment by aspenboy** [comment]
> I tried this one but couldn't get it working with expect(). Do you have an example of this?

It's a separate assertion mechanism, so you don't use it within `expect()`. Pretty nice examples are included on the github page: https://github.com/okken/pytest-check

Also - you can mix assertions in your tests, so it's fine to use `assert`, `expect()` and `check()`.

**Comment by mbroton** [comment]
I agree, it would be really useful to have soft assertions in Python version of Playwright. Right now I'm using workarounds in some of my projects, similar to what was already mentioned above.

I spent some time exploring the code and came up with an approach that seems to work. Here’s an example of how it looks:
```python
def test_foo(page: Page) -> None:
    # ...
    with soft_assertions() as expect:
        expect(page.locator("div")).to_have_text("foo")
        expect(page.locator("div")).to_have_text("bar")
```
`expect` in this example works just like a normal `expect` (works with pages, locators and api responses).

At first, I considered an `expect.soft(...)` approach like in Node, but I couldn’t find a solution that didn’t require a lot of changes.

Would love any feedback on this!

**Comment by Skn0tt** [comment]
Let's also try out implementing it via a fixture in our pytest integration.
